mobile/lib/features/trade/candlestick_chart.dart

```diff
@@ -1,14 +1,78 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:candlesticks/candlesticks.dart';
 import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
 
-class CandlestickChart extends StatelessWidget {
+class CandlestickChart extends StatefulWidget {
   const CandlestickChart({super.key});
 
+  @override
+  State<CandlestickChart> createState() => _CandlestickChartState();
+}
+
+class _CandlestickChartState extends State<CandlestickChart> {
+  List<Candle> candles = [];
+  bool themeIsDark = false;
+  Timer? timer;
+
+  @override
+  void initState() {
+    fetchCandles().then((value) {
+      setState(() {
+        candles = value;
+      });
+    });
+    super.initState();
+    timer = Timer.periodic(const Duration(seconds: 30), (Timer t) => getNewCandles());
+  }
+
+  Future<List<Candle>> fetchCandles() async {
+    final uri = Uri.parse(
+        "https://www.bitmex.com/api/v1/trade/bucketed?binSize=1m&partial=false&symbol=XBTUSD&count=1000&reverse=true");
+    final res = await http.get(uri);
+    return (jsonDecode(res.body) as List<dynamic>).map((e) => parse(e)).toList().toList();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return const SizedBox(
-        height: 200,
-        child: DecoratedBox(
-            decoration: BoxDecoration(color: Colors.grey),
-            child: Center(child: Text("Candlestick Chart"))));
+    return SizedBox(
+      height: 300,
+      child: Candlesticks(
+        candles: candles,
+      ),
+    );
+  }
+
+  Future<void> getNewCandles() async {
```

Comment:
This function should not be in the `widget` code, but should be move to a `Service` - we do fetch data from an external system here.

It would be great if the service is configurable in the widget - that way it can be mocked in the test (that should answer how to handle this widget in the test :)